packages/react/src/schematics/component/component.ts

```diff
@@ -0,0 +1,135 @@
+import { join, Path } from '@angular-devkit/core';
+import * as ts from 'typescript';
+import {
+  apply,
+  chain,
+  filter,
+  mergeWith,
+  move,
+  noop,
+  Rule,
+  template,
+  Tree,
+  url
+} from '@angular-devkit/schematics';
+import { Schema } from './schema';
+import { names } from '@nrwl/workspace';
+import {
+  addDepsToPackageJson,
+  addGlobal,
+  getProjectConfig,
+  insert
+} from '@nrwl/workspace/src/utils/ast-utils';
+import { reduceCssInJsModules, getCssInJsModule } from '../../utils/css-in-js';
+import { PackageDependencies } from '../../utils/dependencies';
+import { formatFiles } from '@nrwl/workspace';
+
+interface NormalizedSchema extends Schema {
+  cssInJsModule: null | string;
+  extraPackageDependencies: null | PackageDependencies;
+  projectSourceRoot: Path;
+  fileName: string;
+  className: string;
+}
+
+export default function(schema: Schema): Rule {
+  return (host: Tree) => {
+    const options = normalizeOptions(host, schema);
+    return chain([
+      createComponentFiles(options),
+      addPackageDependencies(options),
+      addExportsToBarrel(options),
+      formatFiles({ skipFormat: false })
+    ]);
+  };
+}
+
+function createComponentFiles(options: NormalizedSchema): Rule {
+  return mergeWith(
+    apply(url(`./files`), [
+      template({
+        ...options,
+        tmpl: ''
+      }),
+      options.skipTests ? filter(file => !/.*spec.tsx/.test(file)) : noop(),
+      options.cssInJsModule
+        ? filter(file => !file.endsWith(`.${options.style}`))
+        : noop(),
+      move(join(options.projectSourceRoot, 'lib'))
+    ])
+  );
+}
+
+function addPackageDependencies(options: NormalizedSchema): Rule {
+  return options.extraPackageDependencies
+    ? addDepsToPackageJson(
+        options.extraPackageDependencies.dependencies,
+        options.extraPackageDependencies.devDependencies
+      )
+    : noop();
+}
+
+function addExportsToBarrel(options: NormalizedSchema): Rule {
+  return options.export
+    ? (host: Tree) => {
+        const indexFilePath = join(options.projectSourceRoot, 'index.ts');
+        const buffer = host.read(indexFilePath);
+
+        if (!!buffer) {
+          const indexSource = buffer!.toString('utf-8');
+          const indexSourceFile = ts.createSourceFile(
+            indexFilePath,
+            indexSource,
+            ts.ScriptTarget.Latest,
+            true
+          );
+
+          insert(
+            host,
+            indexFilePath,
+            addGlobal(
+              indexSourceFile,
+              indexFilePath,
+              `export { default as ${options.className}, ${options.className}Props } from './lib/${
+                options.name
+              }/${options.fileName}';`
+            )
+          );
+        }
+
+        return host;
+      }
+    : noop();
+}
+
+function normalizeOptions(host: Tree, options: Schema): NormalizedSchema {
+  let extraPackageDependencies = null;
```

Comment:
Hm, I don't think we need to add this to the options object. We can calculate it when it is needed right?